src/execution_plan/ops/op_node_by_label_scan.c

```diff
@@ -77,23 +130,30 @@ static Record NodeByLabelScanConsumeFromChild(OpBase *opBase) {
 		if(op->child_record == NULL) return NULL;
 		// One-time construction of the iterator.
 		// (Don't do so before now because the label might have been built in a child op.)
-		if(op->iter == NULL) _ConstructIterator(op);
-		else GxB_MatrixTupleIter_reset(op->iter);
+		if(op->iter == NULL) {
+			if(!_ConstructIterator(op)) {
+				// No need to iterate - consume child.
+				while(OpBase_Consume(op->op.children[0])) {}
+				return NULL;
+			}
+		} else {
+			_ResetIterator(op);
+		}
 	}
 
 	GrB_Index nodeId;
 	bool depleted = false;
 	GxB_MatrixTupleIter_next(op->iter, NULL, &nodeId, &depleted);
-	if(depleted) {
+	if(depleted || _outOfBounds(op, nodeId)) {
```

Comment:
If you like we can set the iterator `nnz_idx` to A->p[maxRowIdx];` that way we can have a range bounded iterator, currently we only have a start with an open end iterator.

Reply:
done